src/main/java/com/ue/shopsystem/impl/PlayershopImpl.java

```diff
@@ -197,7 +197,8 @@ public Inventory getStockpileInventory() {
 	 */
 
 	private void updateItemInStockpile(int slot) throws GeneralEconomyException, ShopSystemException {
-		ItemStack stack = getShopInventory().getItem(slot).clone();
+		ItemStack stack = null;
+		if (getShopInventory().getItem(slot) != null) stack = getShopInventory().getItem(slot).clone();

```

Comment:
The reason for this null check is because the code was crashing on a NullPointerException when it attempted to load inventories. This happened consistently at the line
stack = getShopInventory()getItem(slot).clone();
After spending a few hours working through the problem I came to the conclusion that by calling the .clone() function on Object.clone() in this way passing the null value to this function specifically would throw the NPE. The local object <ItemStack stack> is still used throughout all of the code with a default assignment of null. The only thing that has changed is now it checks whether or not the item in the inventory slot is null or not before attempting to clone it. If the inventory slot is null even though it defaults to Material.AIR it doesn't actually contain any data that needs to be cloned anyway.